src/test/functional/pages/BuzzPage.ts

```diff
@@ -1,66 +1,194 @@
-    public readonly resharedTitleText = this.page.locator('.orangehrm-buzz-post-body-original-text')
-
-
-    public async locateElementWithDynamicTextAndPhoto(randomTitle: string) {
-      const element = this.postBody.locator(`:has(:text('${randomTitle}')):has(.orangehrm-buzz-photos) .oxd-text.oxd-text--p.orangehrm-buzz-post-body-text`)
-      return element;
+ 
+    public getMostLikedTab(
+      ): Locator {
+        return this.page.getByRole('button', { name: Labels.MOST_LIKED_POST })
+      }
+    
+    public getMostCommentedTab(
+      ): Locator {
+        return this.page.getByRole('button', { name: Labels.MOST_COMMENTED_POST})
+      }
+    
+    public getTextPostBody(
+      ): Locator {
+        return this.page.locator('.orangehrm-buzz-post-body-text')
+      }
+    
+    public getVideoBody(
+      ): Locator {
+        return this.page.locator('.orangehrm-buzz-video')
+      }
+
+    protected getSubmitSimplePostButton(
+      ): Locator {
+        return this.page.getByRole('button', { name: Labels.POST, exact: true })
+      }
+
+    protected getPostMessageInput(
+      ): Locator {
+        return this.page.getByPlaceholder(Labels.WHAT_ON_YOUR_MIND)
+      }
+    
+    protected getShareVideoTitle(
+      ): Locator {
+        return this.page.getByRole('dialog').getByPlaceholder(Labels.WHAT_ON_YOUR_MIND)
+      }
+    
+    protected getVideoUrlTextarea(
+      ): Locator {
+        return this.page.getByPlaceholder(Labels.PASTE_VIDEO_URL)
+      }
+    
+    protected getShareVideoButton(
+      ): Locator {
+        return this.page.getByRole('button', { name: Labels.SHARE_VIDEO })
+      }
+    
+    protected getConfirmEditedPostButton(
+      ): Locator {
+        return this.page.getByRole('dialog').getByRole('button', { name: Labels.POST })
+      }
+    
+    protected getPostEditTextarea(
+      ): Locator {
+        return this.page.getByRole('dialog').locator('textarea')
+      }
+    
+    protected getConfirmDeleteButton(
+      ): Locator {
+        return this.page.getByRole('button', { name: Labels.YES_DELETE});
+      }
+    
+    protected getEditButton(
+      ): Locator {
+        return this.page.getByText(Labels.EDIT_POST);
+      }
+
+    protected getDletePostButton(
+      ): Locator {
+        return this.page.getByText(Labels.DELETE_POST);
+      }
+
+    public getThreeDotsButton(
+      ): Locator {
+        return this.page.getByRole('button', { name: '' });
+      }
+
+    public getSharingSubmitButton(
+      ): Locator {
+        return this.page.getByRole('button', { name: Labels.SHARE, exact: true })
+      }
+
+    protected getFileInputButton(
+      ): Locator {
+        return this.page.locator('input[type="file"]')
+      }
+
+    protected getSharePhotoButton(
+    ): Locator {
+      return this.page.getByRole('button', { name: Labels.SHARE_PHOTOS })
+    }
+
+    public getCommentInput(
+      randomTitle: string
```

Comment:
I would probably not name this randomTitle but just title. The place where you call this function does not use randomTitle. Even if it would be random from the callers perspective - still using "title" as name makes more sense. Because if you would just pass not a random title, then the parameters name would be more generic.

Reply:
right when we use it lebel.specific_label then it is one lebel not more

Reply:
done